modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java

```diff
@@ -184,17 +184,21 @@ public void archiveOrganization(TenantInfo tenantInfo) throws RuntimeProvisionin
             kubClient = KubernetesProvisioningUtils.getFabric8KubernetesClient();
             DeploymentList deploymentList = kubClient.extensions().deployments().list();
 
-            if (deploymentList.getItems().contains(deployment)) {
+            Deployment currentDeployement = kubClient.inNamespace(namespace.getMetadata().getName()).extensions()
+                    .deployments().withName(config.getDeploymentName()).get();
+            if (currentDeployement != null) {
                 //Redeployment
                 //Deployment recreation should happen after comparing the new Deployment config with
                 // running service configs.
                 kubClient.inNamespace(namespace.getMetadata().getName()).extensions()
                         .deployments().withName(config.getDeploymentName()).replace(deployment);
+
+                //todo need to change service replacement because need to check whether service object is available
                 //Service recreation should happen after comparing the new service config with running service configs.
-                for (Service service : serviceList) {
-                    kubClient.inNamespace(namespace.getMetadata().getName()).services().replace(service);
-                    serviceNameList.add(service.getMetadata().getName());
-                }
+//                for (Service service : serviceList) {
```

Comment:
This is code comment is removed from following commit.
https://github.com/wso2/product-af/pull/800/commits/c707570879201754759203ff99b9c5ea5489dcba